revocation/internal/crl/crl.go

```diff
@@ -148,14 +162,30 @@ func validate(crl *x509.RevocationList, issuer *x509.Certificate) error {
 
 	// unsupported critical extensions is not allowed. (See RFC 5280, Section 5.2)
 	for _, ext := range crl.Extensions {
-		if ext.Critical {
-			return fmt.Errorf("CRL contains unsupported critical extension: %v", ext.Id)
+		switch {
+		case ext.Id.Equal(oidIssuingDistributionPoint):
+			// IssuingDistributionPoint is a critical extension that identifies
+			// the scope of the CRL. Since we will check all the CRL
+			// distribution points, it is not necessary to check this extension.
+		default:
+			if ext.Critical {
+				return fmt.Errorf("CRL contains unsupported critical extension: %v", ext.Id)
+			}
 		}
```

Comment:
nit: question: do we need to check other `ext.Id`? If not, I'd prefer
```suggestion
		if ext.Id.Equal(oidIssuingDistributionPoint) {
			// IssuingDistributionPoint is a critical extension that identifies
			// the scope of the CRL. Since we will check all the CRL
			// distribution points, it is not necessary to check this extension.
			continue
		}
		if ext.Critical {
			return fmt.Errorf("CRL contains unsupported critical extension: %v", ext.Id)
		}
```

Reply:
Updated.